staging/src/k8s.io/apiserver/pkg/storage/cacher/metrics/metrics.go

```diff
@@ -176,6 +176,29 @@ var (
 			Help:           "Counter for consistent reads from cache.",
 			StabilityLevel: compbasemetrics.ALPHA,
 		}, []string{"resource", "success", "fallback"})
+
+	WatchDispatchEventNumWatchers = compbasemetrics.NewHistogramVec(
+		&compbasemetrics.HistogramOpts{
+			Namespace:      namespace,
+			Subsystem:      subsystem,
+			Name:           "watch_dispatch_event_num_watchers",
+			Help:           "Histogram of total number of watchers to dispatch event to.",
+			Buckets:        []float64{1, 2, 3, 4, 6, 8, 10, 15, 20, 30, 40, 50, 100, 200, 300, 500, 800, 1000, 2000, 3000},
+			StabilityLevel: compbasemetrics.ALPHA,
+		},
+		[]string{"resource"},
+	)
+
+	WatcherCountGauge = compbasemetrics.NewGaugeVec(
+		&compbasemetrics.GaugeOpts{
+			Namespace:      namespace,
+			Subsystem:      subsystem,
+			Name:           "watcher_counter",
+			Help:           "Counter of the inflight cacher watchers.",
+			StabilityLevel: compbasemetrics.ALPHA,
+		},
+		[]string{"resource", "index"},
```

Comment:
From what I understand, indexes will be based on labels now on top of the built-in indexes and these can grow exponentially making this particular metric explode in terms of cardinality. Is my understanding correct that the  indexes will be unbounded?

Reply:
I have added a `--index-labels` in `staging/src/k8s.io/apiserver/pkg/server/options/etcd.go` in this PR. So label indexes are configurated by the cluster administrator as apiserver command line options such as `--index-labels=pods#k8s-app;app;spark-app-selector` and they won't be unbounded.

Reply:
I wonder if this is the right approach for your feature. Adding a new cli flag that cluster admins would have to be taught about sounds a bit much for just a metric. Maybe this information could be put somewhere else? For example in logs or in audit logs?